ingestclient/core/backend.py

```diff
@@ -479,6 +479,8 @@ def get_job_status(self, ingest_job_id):
         r = requests.get('{}/{}/ingest/{}/status'.format(self.host, self.api_version, ingest_job_id),
                          headers=self.api_headers, verify=self.validate_ssl)
 
+        if r.text == '{"status": 404, "code": 2004, "message": "The job with id %s has been deleted"}' % (ingest_job_id):
```

Comment:
We should fix the response from the endpoint so it returns a normal 200 with the status set properly instead of a 404. Currently, we don't match our docs when getting status for a deleted job: https://theboss.readme.io/docs/get-ingest-job-status

Then in client.py, you can do a normal `get_job_status()` call.

Fix would be here: https://github.com/jhuapl-boss/boss/blob/08a0ddb86b7dedf1c0b029d9479db0dc2b11ff1a/django/bossingest/views.py#L362